docs/design/2290-cadence-ndc.md

```diff
@@ -0,0 +1,310 @@
+# Design doc: Cadence NDC
+
+Last updated: Dec 2019
+
+Reference: [#2290](https://github.com/uber/cadence/issues/2290)
+
+
+## Abstract
+
+Cadence Cross DC is a feature which asynchronously replicates workflows from active data center to other passive data centers, for reconstruction. When necessary, customer can failover to any of the data centers which has the backup for high availability.
+
+Cadence Cross DC is an AP (in terms of CAP).
+
+This doc explains the high level concepts about Cadence Cross DC (mainly the N data center version).
+
+
+## Newly Introduced Components
+
+1. Version
+2. Version history
+3. Workflow history conflict resolution
+4. Zombie workflows
+5. Workflow task processing
+
+### Version
+
+Version is a newly introduced concept in Cadence Cross DC which describes the chronological order of events (per customer domain).
```

Comment:
Need to explicitly call out that version is attached to domain, not cluster, not workflows.
Adding an example would help as well.

Reply:
added an example